src/main/java/store/domain/Product.java

```diff
@@ -0,0 +1,148 @@
+package store.domain;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import store.dto.ProductsDto;
+
+public class Product {
+
+    private static final String NO_PROMOTION = "null";
+    private static final String PRODUCT_DELIMITER = ",";
+
+    private final String name;
+    private final int price;
+    private int normalStock = 0;
+    private int promotionStock = -1;
+    private Promotion promotion;
+
+    private Product(String name, int price, int stock, String promotion) {
+        this.name = name;
+        this.price = price;
+        classifyNormalOrPromotion(stock, promotion);
+    }
+
+    public static Product createProduct(String productInformation) {
+        List<String> productInformations = new ArrayList<>(Arrays.stream(productInformation.split(PRODUCT_DELIMITER)).toList());
+
+        return new Product(productInformations.get(0),
+                Integer.parseInt(productInformations.get(1)),
+                Integer.parseInt(productInformations.get(2)),
+                productInformations.get(3));
+    }
+
+    public boolean equalsTo(String name) {
+        return this.name.equals(name);
+    }
+
+    public void addNewStock(String productInformation) {
+        List<String> productInformations = new ArrayList<>(Arrays.stream(productInformation.split(",")).toList());
+
+        int stock = Integer.parseInt(productInformations.get(2));
+        String promotion = productInformations.get(3);
+
+        classifyNormalOrPromotion(stock, promotion);
+    }
+
+    public boolean isOverPromotionQuantity(int quantity) {
+        if (isPromotionApply() && promotion.getPromotionApplyQuantity() > quantity) {
+            return false;
+        }
+        return true;
+    }
+
+    public boolean canBuy(int quantity) {
+        if (isPromotionApply()) {
+            return quantity <= normalStock + promotionStock;
+        }
+        return quantity <= normalStock;
+    }
+
+    public int notApplyPromotionOutOfStockCounts(int quantity) {
```

Comment:
동사형 이름을 사용하는게 명확하지 않을까요?
calculateNonPromotionalOutOfStockCount 과 같은 이름은 어떨까 합니다
```suggestion
    public int calculateNonPromotionalOutOfStockCount(int quantity) {
```

Reply:
메서드명은 동사로 시작해야 한다는 것을 프리코스 기간동안 명심했는데.. 결국 놓친 부분이 있었네요🥲 제시해주신 개선 방향이 메서드를 나타내기에 훨씬 좋은 방법 같습니다! 감사합니다😀